TradingBot/Services/Strategy.cs

```diff
@@ -1,33 +1,188 @@
-﻿using Microsoft.AspNetCore.Authentication;
+using Microsoft.EntityFrameworkCore;
+using Alpaca.Markets;
+using TradingBot.Database;
 using TradingBot.Models;
+using Microsoft.AspNetCore.Authentication;
+using SQLitePCL;
+using TradingBot.Migrations;
+using Newtonsoft.Json.Linq;
 
 namespace TradingBot.Services;
 
 public interface IStrategy
 {
-    public Task<IReadOnlyList<TradingAction>> GetTradingActionsAsync();
+    public Task<IReadOnlyList<TradingAction>> GetTradingActionsAsync(CancellationToken token = default);
 }
 
 public sealed class Strategy : IStrategy
 {
     private readonly IAssetsDataSource _assetsDataSource;
-    private readonly ISystemClock _clock;
     private readonly IPricePredictor _predictor;
+    private readonly IMarketDataSource _marketDataSource;
+    private readonly ISystemClock _clock;
+    private readonly IStrategyParametersService _strategyParameters;
 
-    public Strategy(IPricePredictor predictor, IAssetsDataSource assetsDataSource, ISystemClock clock)
+    public Strategy(IPricePredictor predictor, IAssetsDataSource assetsDataSource, IMarketDataSource marketDataSource, 
+        ISystemClock clock, IStrategyParametersService strategyParameters)
     {
         _predictor = predictor;
         _assetsDataSource = assetsDataSource;
+        _marketDataSource = marketDataSource;
         _clock = clock;
+        _strategyParameters = strategyParameters;
     }
 
-    public async Task<IReadOnlyList<TradingAction>> GetTradingActionsAsync()
+    public async Task<IReadOnlyList<TradingAction>> GetTradingActionsAsync(CancellationToken token = default)
+    {
+        var strategyParameters = await _strategyParameters.GetConfigurationAsync();
+        var actions = await DetermineTradingActionsAsync(strategyParameters.MaxStocksBuyCount, strategyParameters.MinDaysDecreasing, strategyParameters.MinDaysIncreasing, strategyParameters.TopGrowingSymbolsBuyRatio, token);
+
+        return actions;
+    }
+
+    private async Task<IReadOnlyList<TradingAction>> DetermineTradingActionsAsync(int maxBuyCount, int minDaysDecreasing, int minDaysIncreasing, decimal topGrowingSymbolsBuyRatio, CancellationToken token = default)
```

Comment:
We could pass `StrategyParameters` here. It might be clearer than having 5 arguments in a method.